index.js

```diff
@@ -0,0 +1,157 @@
+    answerBtnC.value = this.answerC;                                                                      //解答選択肢-3出力用要素の値代入
+    answerBtnD.value = this.answerD;                                                                      //解答選択肢-4出力用要素の値代入
+
+    title.after(categoryNode);                                                                            //各要素の出力
+    category.after(difficultyNode);
+    document.body.appendChild(answerBtnA);
+    document.body.appendChild(answerBtnB);
+    document.body.appendChild(answerBtnC);
+    document.body.appendChild(answerBtnD);
+
+    btnA.addEventListener('click', () => {                                                                //正解解答選択肢押下時処理
+      correctAnswerCount++;                                                                                //正解回答数インクリメント
+    });
+  }
+}
+window.addEventListener('load', () => {                                                                  //画面ロード後の処理
+  const startQuiz = document.getElementById('start-quiz');                                                //開始ボタンノード取得
+  const title = document.getElementById('title');                                                         //タイトルノード取得
+  const operation = document.getElementById('operation');                                                 //問題文出力用ノード取得
+  let count = 0;                                                                                          //取得クイズカウント用変数宣言
+  let questionNumber = 1;                                                                                 //クイズNo.格納用変数宣言
+  startQuiz.addEventListener('click', main);                                                              //開始ボタン押下時処理
+
+  function main() {                                                                                       //メイン関数実行
+    const promise =                                                                                        //非同期処理実行
+      new Promise((resolve, reject) => {                                                                    //待機処理
+        title.textContent = '取得中';                                                                        //タイトルノード文字列修正
+        operation.textContent = '少々お待ちください';                                                        //質問文出力用ノード文字列修正
+        startQuiz.remove();                                                                                  //開始ボタン削除
+        setTimeout(resolve, 3000);                                                                           //タイマー関数-3000ms後、満足処理実行
+      })
+      .then(() => {                                                                                         //満足処理
+        sendApiRequest();                                                                                    //APIリクエスト送信関数実行
+      })
+      .catch(() => {                                                                                        //拒絶処理
+        console.log('rejectされました');
+      })
+  }
```

Comment:
- Promise を使うことは問題ないのですが `async/await` を使った方がよりコードが簡潔になります
- `setTimeout(resolve, 3000);` は不要です
  - Promise や `async/await` を使えば非同期処理であってもちゃんと「上から順番に実行される」コードになるので、処理をタイマーで待つ必要はありません

```suggestion
  startQuiz.addEventListener('click', async () => {
    try {
      title.textContent = '取得中';
      operation.textContent = '少々お待ちください';
      startQuiz.remove();
      await sendApiRequest();
    } catch (e) {
      console.error(`rejectされました。理由 => ${e}`);
    }
  });
});
```